tremor-script/src/interpreter/imut_expr.rs

```diff
@@ -60,12 +62,37 @@ where
         meta: &'run Value<'event>,
         local: &'run LocalStack<'event>,
     ) -> Result<Cow<'event, str>> {
-        match stry!(self.run(opts, env, event, state, meta, local)).borrow() {
-            Value::String(s) => Ok(s.clone()),
-            other => error_need_obj(self, self, other.value_type(), &env.meta),
+        match stry!(self.run(opts, env, event, state, meta, local)).into_owned() {
```

Comment:
I think this is worth refactoring especially to move the into_owned into the inner so we don't clone potentially large structs.

i.e. if for bad reasons the `run` evaluates to the entire event we'd clone the whole thing before we error while we don't need to.

```rust
let value = stry!(self.run(opts, env, event, state, meta, local));
if let Some(s) = value.as_str() {
  Ok(s.into_owned())
} else {
  error_need_str(self, self, value.value_type(), &env.meta)
}
```

That would also get rid of  the deconstruction and depending on the enum representation.

Reply:
I had presumed that type errors are rare, and never on a "hot" path. I'll keep the performance impact on the error branches in mind. Thanks.

It's a bit sad that `eval_to_string` returns `Result<Cow<'event, str>>` instead of `Result<Cow<'run, str>>`, but from its use-sites it's clear that that's probably reasonable. Still, that means that `eval_to_string` _always_ returns an owned copy of the string, instead of being able to propagate the (possibly borrowed) result of `run`.

Side-note: the code in the comment doesn't seem to work as-is: in `Ok(s.into_owned())`, `s` is of type `&str`, and no implementation of `str::into_owned() -> Cow<'event, str>` exists. I'll adapt it slightly to
```rust
        let value = stry!(self.run(opts, env, event, state, meta, local));
        if let Some(s) = value.as_str() {
            Ok(Cow::from(s.to_owned())) // <-- changed
        } else {
            error_need_str(self, self, value.value_type(), &env.meta)
        }
```

Reply:
Ja was written from memory :) tweaks are likely needed.